### PR TITLE
docs: architecture constraints and glossary of arc42

### PR DESCRIPTION
**Pull request by JonasKellerer**
Resolves: #140 

Modeled after https://docs.arc42.org/section-2/
Happy to see your comments.

### CODE REVIEW

File: docs/arc42/02_architecture_constraints.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+- Versioned by git.
+- Input files for SILO are provided in a specific format:
+    - metadata and quality control
+    - sequence data (non-aligned nucleotide sequences, aligned nucleotide sequences, aligned AA sequences )
+    - reference genome
+- LAPIS is backed by a database that understands SILO queries.
+- The system is designed to have the best possible performance. This mostly targets SILO, but also in LAPIS, we have to
+  keep in mind that we are dealing with potentially large data that we have to serve to the client.
+- The target platforms, where SILO and LAPIS are run on, are cloud platforms. We provide executable that can be run on
+  Linux and macOS. Windows platforms are out of scope. (constraints RAM, HDD, etc.?, cloud?)
```
Comment (JonasKellerer) [inline docs/arc42/02_architecture_constraints.md:16]: Do we know, what are the constraints on the RAM, HDD?

File: docs/arc42/12_glossary.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Glossary
+
+| Term    | Definition                                                                                                                                                                                                                                                       |
+|---------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| AA      | amino acid                                                                                                                                                                                                                                                       |
+| aligned | a nucleotide sequence is aligned, if it is arranged such that it has many similarities to a given reference genome. The aligned sequence has the same length as the reference genome. Gaps are marked in the aligned sequence. Insertions are stored separately. |
```
Comment (chaoran-chen) [inline docs/arc42/12_glossary.md:6]: Add:

- QC: quality control; in our case, it usually refers to the quality checks and metrics of the sequences
- Variant: We follow a very open definition of variants. Every subset of sequences is considered a variant. A variant is specified by lineage/clade names and mutations. A variant does not need to be [monophyletic](https://en.wikipedia.org/wiki/Monophyly).

File: docs/arc42/02_architecture_constraints.md (hunk 1 above)
Comment (chaoran-chen) [reply docs/arc42/02_architecture_constraints.md:16]: Maybe mention that we don't plan to provide executables for Windows but through Docker, it is of course still possible to run it on a Windows system.

**Review by chaoran-chen — CHANGES_REQUESTED** [review]
Thanks for the draft!

**Comment by fengelniederhammer** [comment]
What do we do with this PR? Incorporate the contents into the wiki and delete `docs/arc42/`?

**Comment by JonasKellerer** [comment]
I think incoproptating and deleting is the way forward, so we only have one place of documentation.

**Comment by chaoran-chen** [comment]
yes, I agree!

**Comment by fengelniederhammer** [comment]
Was part of https://github.com/GenSpectrum/LAPIS/pull/531

**Pull request closed without merging**